From 068aaa979534f0049a985071dbe5724b20bf09d4 Mon Sep 17 00:00:00 2001
From: Peter Krempa <pkrempa@redhat.com>
Date: Mon, 7 Feb 2022 12:42:34 +0100
Subject: qemu: process: Move call to qemuProcessRefreshCPU after cpu probe
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Similarly to previous commit we need to probe the vcpus first.

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/libvirt/+bug/1982896
Origin: upstream, https://gitlab.com/libvirt/libvirt/-/commit/068aaa979534f0049a985071dbe5724b20bf09d4
Last-Update: 2023-08-04
---
diff --git a/src/qemu/qemu_process.c b/src/qemu/qemu_process.c
index f99b14b846..bd9c6ed747 100644
--- a/src/qemu/qemu_process.c
+++ b/src/qemu/qemu_process.c
@@ -8850,14 +8850,14 @@ qemuProcessReconnect(void *opaque)
     ignore_value(qemuSecurityCheckAllLabel(driver->securityManager,
                                            obj->def));
 
-    if (qemuProcessRefreshCPU(driver, obj) < 0)
-        goto error;
-
     if (qemuDomainRefreshVcpuInfo(driver, obj, QEMU_ASYNC_JOB_NONE, true) < 0)
         goto error;
 
     qemuDomainVcpuPersistOrder(obj->def);
 
+    if (qemuProcessRefreshCPU(driver, obj) < 0)
+        goto error;
+
     if (qemuDomainUpdateMemoryDeviceInfo(driver, obj, QEMU_ASYNC_JOB_NONE) < 0)
         goto error;
 
